### PR TITLE
Misc python 2 cleanup, mostly comments and docstrings

### PR DESCRIPTION
**Pull request by peterjc**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/HMM/MarkovModel.py [hunk 1]
```diff
@@ -667,17 +667,11 @@ def _log_transform(self, probability):
...
-            # On Python 2.5 or older that was handled in C code,
-            # and failed on Windows XP 32bit
-            neg_inf = -1e400
         for key in log_prob:
             prob = log_prob[key]
             if prob > 0:
                 log_prob[key] = math.log(log_prob[key])
             else:
-                log_prob[key] = neg_inf
+                log_prob[key] = float("-inf")
```
Comment (mdehoon) [inline Bio/HMM/MarkovModel.py]: You can use `-math.inf` here.

**Comment by widdowquinn** [comment]
I'm happy with this.

**Comment by peterjc** [comment]
Thanks, rebasing and with infinity change...

**Comment by peterjc** [comment]
Infinity change made, should be OK to apply now.

**Comment by peterjc** [comment]
Thanks.

**Pull request merged**